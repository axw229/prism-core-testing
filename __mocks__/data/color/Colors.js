// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
