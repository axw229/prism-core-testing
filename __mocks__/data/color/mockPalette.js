const mockPalette = {
  'colors': [
    {
      'colorNumber': '6765',
      'coordinatingColors': {
        'coord2ColorId': '11194',
        'coord1ColorId': '2870',
        'whiteColorId': '2162'
      },
      'description': [
        'Fairly Colorful',
        'Moderately Colorful',
        'Pale'
      ],
      'id': '2452',
      'isExterior': true,
      'isInterior': true,
      'name': 'Spa',
      'lrv': 64.476,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Blue'
      ],
      'brandKey': 'SW',
      'red': 167,
      'green': 220,
      'blue': 220,
      'hue': 0.5,
      'saturation': 0.43089430894308944,
      'lightness': 0.7588235294117647,
      'hex': '#a7dcdc',
      'isDark': false,
      'storeStripLocator': '164-C2',
      'similarColors': [
        '2634',
        '2630',
        '11225',
        '2472',
        '2445',
        '2622',
        '2459',
        '1606',
        '11226',
        '2451'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '6860',
      'coordinatingColors': {
        'coord2ColorId': '11339',
        'coord1ColorId': '11364',
        'whiteColorId': '2681'
      },
      'description': [
        'Soft',
        'Fairly Bright'
      ],
      'id': '2547',
      'isExterior': true,
      'isInterior': true,
      'name': 'Eros Pink',
      'lrv': 18.874,
      'brandedCollectionNames': [
        '2020 Play',
        '2019 Naturalist'
      ],
      'colorFamilyNames': [
        'Red'
      ],
      'brandKey': 'SW',
      'red': 200,
      'green': 79,
      'blue': 104,
      'hue': 0.9655647382920111,
      'saturation': 0.5238095238095238,
      'lightness': 0.5470588235294118,
      'hex': '#c84f68',
      'isDark': false,
      'storeStripLocator': '105-C5',
      'similarColors': [
        '2273',
        '2266',
        '2542',
        '2546',
        '1998',
        '2280',
        '2530',
        '2287',
        '2279',
        '30193'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9018',
      'coordinatingColors': {
        'coord2ColorId': '11218',
        'coord1ColorId': '11200',
        'whiteColorId': '2366'
      },
      'description': [
        'Intense',
        'Pale'
      ],
      'id': '11195',
      'isExterior': false,
      'isInterior': true,
      'name': 'Honey Bees',
      'lrv': 70.14,
      'brandedCollectionNames': [
        '2018 Unity'
      ],
      'colorFamilyNames': [
        'Yellow'
      ],
      'brandKey': 'SW',
      'red': 251,
      'green': 214,
      'blue': 130,
      'hue': 0.11570247933884299,
      'saturation': 0.937984496124031,
      'lightness': 0.7470588235294118,
      'hex': '#fbd682',
      'isDark': false,
      'storeStripLocator': '134-C4',
      'similarColors': [
        '2382',
        '11199',
        '11196',
        '2588',
        '2375',
        '11198',
        '11194',
        '11191',
        '3146',
        '2361'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9007',
      'coordinatingColors': {
        'coord2ColorId': '1666',
        'coord1ColorId': '2686',
        'whiteColorId': '2331'
      },
      'description': [
        'Rich',
        'Vibrant',
        'Pale'
      ],
      'id': '11184',
      'isExterior': false,
      'isInterior': true,
      'name': 'Melón Meloso',
      'lrv': 55.314,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Orange'
      ],
      'brandKey': 'SW',
      'red': 242,
      'green': 184,
      'blue': 140,
      'hue': 0.0718954248366013,
      'saturation': 0.7968749999999999,
      'lightness': 0.7490196078431373,
      'hex': '#f2b88c',
      'isDark': false,
      'storeStripLocator': '121-C3',
      'similarColors': [
        '2334',
        '2983',
        '2327',
        '11188',
        '2326',
        '2045',
        '2038',
        '3012',
        '2341',
        '2031'
      ],
      'ignore': false,
      'archived': false
    }
  ],
  'activeColor': {
    'colorNumber': '9007',
    'coordinatingColors': {
      'coord2ColorId': '1666',
      'coord1ColorId': '2686',
      'whiteColorId': '2331'
    },
    'description': [
      'Rich',
      'Vibrant',
      'Pale'
    ],
    'id': '11184',
    'isExterior': false,
    'isInterior': true,
    'name': 'Melón Meloso',
    'lrv': 55.314,
    'brandedCollectionNames': [],
    'colorFamilyNames': [
      'Orange'
    ],
    'brandKey': 'SW',
    'red': 242,
    'green': 184,
    'blue': 140,
    'hue': 0.0718954248366013,
    'saturation': 0.7968749999999999,
    'lightness': 0.7490196078431373,
    'hex': '#f2b88c',
    'isDark': false,
    'storeStripLocator': '121-C3',
    'similarColors': [
      '2334',
      '2983',
      '2327',
      '11188',
      '2326',
      '2045',
      '2038',
      '3012',
      '2341',
      '2031'
    ],
    'ignore': false,
    'archived': false
  },
  'toggleCompareColor': false,
  'temporaryActiveColor': null
}

export default mockPalette
