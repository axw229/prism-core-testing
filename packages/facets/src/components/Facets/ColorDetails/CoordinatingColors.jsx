// @flow
import React, { useContext } from 'react'
import { useSelector } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import ConfigurationContext, { type ConfigurationContextType } from 'src/contexts/ConfigurationContext/ConfigurationContext'
import * as GA from 'src/analytics/GoogleAnalytics'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'
import 'src/scss/convenience/visually-hidden.scss'

const baseClass = 'color-info'
const orderedCoordColorProps = [
  'coord1ColorId',
  'coord2ColorId',
  'coord3ColorId',
  'whiteColorId'
]

type Props = { color: Color, onColorChanged: Color => void }
function CoordinatingColors ({ color, onColorChanged }: Props) {
  const { brandId, colorWall: { colorSwatch = {} } }: ConfigurationContextType = useContext(ConfigurationContext)
  const { colorNumOnBottom = false, houseShaped = false } = colorSwatch
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)
  const { coordinatingColors } = color
  const btnClass = houseShaped ? `${baseClass}-house-shaped` : baseClass

  return (
    <>
      <h5 className='visually-hidden'><FormattedMessage id='COORDINATING_COLORS' /></h5>
      <ul className={`${baseClass}__coord-colors`}>
        {orderedCoordColorProps.map(prop => coordinatingColors && coordinatingColors[prop]).filter(Boolean).map(id => colorMap[id]).filter(Boolean).map((color: Color) => {
          return (
            <li
              key={color.colorNumber}
              className={`${baseClass}__coord-color ${color.isDark ? `${baseClass}__coord-color--dark-color` : ''}`}
              style={{ backgroundColor: color.hex }}
            >
              <button
                className={`${baseClass}__color-swatch-link ${btnClass}${colorNumOnBottom ? '__name-number' : '__number-name'}`}
                onClick={() => {
                  onColorChanged(color)
                  GA.event({ category: 'Color Detail / Coordinating Color', action: 'View Coord Color', label: color.name }, GA_TRACKER_NAME_BRAND[brandId])
                }}
              >
                <p className={`${btnClass}__coord-color-number`}>
                  {`${color.brandKey} ${color.colorNumber}`}
                </p>
                <p className={`${btnClass}__coord-color-name`}>
                  {color.name}
                </p>
              </button>
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default CoordinatingColors
