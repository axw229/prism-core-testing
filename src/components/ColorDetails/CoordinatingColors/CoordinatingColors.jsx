// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { find } from 'lodash'

import { paintAllSceneSurfaces } from '../../../actions/scenes'

type Props = {
  colors: Array,
  color: Object,
  paintAllSceneSurfaces: Function
}

class CoordinatingColors extends PureComponent<Props> {
  render () {
    const coordinatingColors = this.coordinatingColors()

    return (
      <React.Fragment>
        <strong>Coordinating Colors</strong>
        <hr />
        <ul>
          <li><button onClick={() => this.selectColor(coordinatingColors.coord1Color)}>{coordinatingColors.coord1Color.name}</button></li>
          <li><button onClick={() => this.selectColor(coordinatingColors.coord2Color)}>{coordinatingColors.coord2Color.name}</button></li>
          <li><button onClick={() => this.selectColor(coordinatingColors.whiteColor)}>{coordinatingColors.whiteColor.name}</button></li>
        </ul>
        <hr />
      </React.Fragment>
    )
  }

  coordinatingColors () {
    const { colors, color } = this.props
    const coord1ColorId = color.coordinatingColors.coord1ColorId
    const coord2ColorId = color.coordinatingColors.coord2ColorId
    const whiteColorId = color.coordinatingColors.whiteColorId

    const coord1Color = find(colors, color => color.id == coord1ColorId) // eslint-disable-line
    const coord2Color = find(colors, color => color.id == coord2ColorId) // eslint-disable-line
    const whiteColor = find(colors, color => color.id == whiteColorId) // eslint-disable-line

    const coordinatingColors = {
      coord1Color,
      coord2Color,
      whiteColor
    }

    return coordinatingColors
  }

  selectColor (color) {
    this.props.paintAllSceneSurfaces(color)
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    paintAllSceneSurfaces: (color) => {
      dispatch(paintAllSceneSurfaces(color))
    }
  }
}

export default connect(null, mapDispatchToProps)(CoordinatingColors)
