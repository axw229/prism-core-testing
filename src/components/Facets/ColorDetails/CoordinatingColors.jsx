// @flow
import React from 'react'
import { useSelector } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import * as GA from 'src/analytics/GoogleAnalytics'
import 'src/scss/convenience/visually-hidden.scss'

const orderedCoordColorProps = [
  'coord1ColorId',
  'coord2ColorId',
  'whiteColorId'
]

type Props = { color: Color, onColorChanged: Color => void }
function CoordinatingColors ({ color, onColorChanged }: Props) {
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)
  const { coordinatingColors } = color

  return (
    <>
      <h5 className='visually-hidden'><FormattedMessage id='COORDINATING_COLORS' /></h5>
      <ul className={`color-info__coord-colors`}>
        {orderedCoordColorProps.map(prop => coordinatingColors && coordinatingColors[prop]).filter(Boolean).map(id => colorMap[id]).filter(Boolean).map((color: Color) => {
          return (
            <li
              key={color.colorNumber}
              className={`color-info__coord-color ${color.isDark ? `color-info__coord-color--dark-color` : ''}`}
              style={{ backgroundColor: color.hex }}
            >
              <button
                className='color-info__color-swatch-link'
                onClick={() => {
                  onColorChanged(color)
                  GA.event({ category: 'Color Detail / Coordinating Color', action: 'View Coord Color', label: color.name })
                }}
              >
                <p className='color-info__coord-color-number'>
                  {`${color.brandKey} ${color.colorNumber}`}
                </p>
                <p className='color-info__coord-color-name'>
                  {color.name}
                </p>
              </button>
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default CoordinatingColors
