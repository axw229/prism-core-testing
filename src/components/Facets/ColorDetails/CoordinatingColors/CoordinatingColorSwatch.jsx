// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'

import { paintAllSceneSurfaces } from '../../../../actions/scenes'

type Props = {
  color: Object,
  paintAllSceneSurfaces: Function
}

class CoordinatingColorSwatch extends PureComponent<Props> {
  static baseClass = 'color-info'

  constructor (props) {
    super(props)

    this.selectColor = this.selectColor.bind(this)
  }

  render () {
    const { color } = this.props

    // not all colors have all coordinating colors
    if (!color) {
      return null
    }

    return (
      <li className={`${CoordinatingColorSwatch.baseClass}__coord-color`} onClick={this.selectColor} style={{ backgroundColor: color.hex }}>
        <p className={`${CoordinatingColorSwatch.baseClass}__coord-color-number`}>
          {`${color.brandKey} ${color.colorNumber}`}
        </p>
        <p className={`${CoordinatingColorSwatch.baseClass}__coord-color-name`}>{color.name}</p>
      </li>
    )
  }

  selectColor = function selectColor () {
    this.props.paintAllSceneSurfaces(this.props.color)
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    paintAllSceneSurfaces: (color) => {
      dispatch(paintAllSceneSurfaces(color))
    }
  }
}

export default connect(null, mapDispatchToProps)(CoordinatingColorSwatch)
