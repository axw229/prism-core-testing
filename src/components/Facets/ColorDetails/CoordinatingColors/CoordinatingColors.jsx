// @flow
import React from 'react'
import { FormattedMessage } from 'react-intl'
import find from 'lodash/find'

import type { Color, ColorMap } from '../../../../shared/types/Colors'

import CoordinatingColorSwatch from './CoordinatingColorSwatch'

import 'src/scss/convenience/visually-hidden.scss'

type Props = {
  colors: ColorMap,
  color: Color
}

function CoordinatingColors ({ colors, color }: Props) {
  const coord1ColorId = color.coordinatingColors.coord1ColorId
  const coord2ColorId = color.coordinatingColors.coord2ColorId
  const whiteColorId = color.coordinatingColors.whiteColorId

  const coord1Color = find(colors, color => color.id === coord1ColorId)
  const coord2Color = find(colors, color => color.id === coord2ColorId)
  const whiteColor = find(colors, color => color.id === whiteColorId)

  return (
    <React.Fragment>
      <h5 className='visually-hidden'><FormattedMessage id='COORDINATING_COLORS' /></h5>
      <ul className={`color-info__coord-colors`}>
        <CoordinatingColorSwatch color={coord1Color} />
        <CoordinatingColorSwatch color={coord2Color} />
        <CoordinatingColorSwatch color={whiteColor} />
      </ul>
    </React.Fragment>
  )
}

export default CoordinatingColors
