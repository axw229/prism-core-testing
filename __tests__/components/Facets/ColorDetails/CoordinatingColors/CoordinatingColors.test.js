/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import CoordinatingColors from 'src/components/Facets/ColorDetails/CoordinatingColors/CoordinatingColors'
import * as Colors from '__mocks__/data/color/Colors'
import CoordinatingColorSwatch from 'src/components/Facets/ColorDetails/CoordinatingColors/CoordinatingColorSwatch'
import { FormattedMessage } from 'react-intl'

const color = Colors.getColor()
const colors = Colors.getAllColors()

const getCoordinatingColors = (props) => {
  return shallow(<CoordinatingColors {...props} />)
}

describe('CoordinatingColors component with props', () => {
  let coordinatingColors
  beforeEach(() => {
    if (!coordinatingColors) {
      coordinatingColors = getCoordinatingColors({ colors: colors, color: color })
    }
  })

  it('CoordinatingColors is rendering', () => {
    expect(coordinatingColors.exists()).toBe(true)
  })

  it('CoordinatingColors is rendering H5 with FormattedMessage', () => {
    expect(coordinatingColors.find('h5.visually-hidden').contains(<FormattedMessage id='COORDINATING_COLORS' />)).toBeTruthy()
  })

  it('CoordinatingColors is rendering CoordinatingColorSwatch under ul', () => {
    expect(coordinatingColors.find('ul.color-info__coord-colors').contains(<CoordinatingColorSwatch />)).toBeTruthy()
  })
})
