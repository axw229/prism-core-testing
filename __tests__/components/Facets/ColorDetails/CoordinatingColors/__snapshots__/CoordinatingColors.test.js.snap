// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoordinatingColors component with props CoordinatingColors Snapshot with props 1`] = `
<Fragment>
  <h5
    className="visually-hidden"
  >
    <FormattedMessage
      id="COORDINATING_COLORS"
      values={Object {}}
    />
  </h5>
  <ul
    className="color-info__coord-colors"
  >
    <CoordinatingColorSwatch />
    <CoordinatingColorSwatch />
    <CoordinatingColorSwatch />
  </ul>
</Fragment>
`;
